go/test/endtoend/vtgate/reservedconn/sysvar_test.go

```diff
@@ -427,18 +427,36 @@ func TestSysVarTxIsolation(t *testing.T) {
 	require.NoError(t, err)
 	defer conn.Close()
 
+	// will run every check twice to see that the isolation level is set for all the queries in the session and
+
 	// default from mysql
 	utils.AssertMatches(t, conn, "select @@transaction_isolation", `[[VARCHAR("REPEATABLE-READ")]]`)
+	// ensuring it goes to mysql
+	utils.AssertContains(t, conn, "select @@transaction_isolation, connection_id()", `REPEATABLE-READ`)
+	// second run, to ensuring the setting is applied on the session and not just on next query after settings.
+	utils.AssertContains(t, conn, "select @@transaction_isolation, connection_id()", `REPEATABLE-READ`)
 
 	// setting to different value.
 	utils.Exec(t, conn, "set @@transaction_isolation = 'read-committed'")
-	utils.AssertMatches(t, conn, "select @@transaction_isolation", `[[VARCHAR("READ-COMMITTED")]]`)
+	utils.AssertMatches(t, conn, "select @@transaction_isolation", `[[VARCHAR("read-committed")]]`)
```

Comment:
Why are some values uppercase and some lowercase?
Line 437 - `REPEATABLE-READ`. But everything else is lowercase.

Reply:
Looks like it may be an artifact of how the test input is provided. To avoid confusion, my suggestion is to make them all uniform.

Reply:
I will raise a separate PR to handle it.